server/src/main/java/org/elasticsearch/snapshots/SnapshotInfo.java

```diff
@@ -54,6 +54,7 @@
     private static final DateFormatter DATE_TIME_FORMATTER = DateFormatter.forPattern("strictDateOptionalTime");
     private static final String SNAPSHOT = "snapshot";
     private static final String UUID = "uuid";
+    private static final String REPOSITORY = "repository";
```

Comment:
@original-brownbear I don't see how adding repository field to `SnapshotInfo` is harmful. 
Yes, I see that `BlobStoreRepository` writes `SnapshotInfo` to disk and reads it, but it does not seem to be a problem if the repository name is immutable.
I find it natural that `SnapshotInfo`, which has the following JavaDoc `Information about a snapshot` contains information about the repository name for which this snapshot was taken.
At the same time, I'm not a proponent of adding unnecessary classes unless we can come up with a good argument why current implementation won't work.

Reply:
It simply seems wrong and needless to go through all the steps of adding BwC, changing the format of what we serialize in the snapshot repository (which always introduces risk) etc. when we can resolve this trivially by simply changing the response serialization.
It would even be a smaller change.
=> I really don't see an argument for changing the serialization in the repo or over the wire when we already can get all the data we need inside the transport action. All we do with this change is store redundant data in the repo, add risk and BwC issues when we can have the same change without those downsides by simply fixing the response serialization.

Not adding a new class here to keep complexity down isn't really an argument either because the whole logic to figure out the state of things is so complex precisely because we are sharing `SnapshotInfo`  across functionalities here (sorta related to https://github.com/elastic/elasticsearch/pull/40943#issuecomment-488664150).